### PR TITLE
ds18b20: device driver, work with onewire_bus library (BSP-319)

### PR DESCRIPTION
**Pull request by suda-morris**
Added the driver for the DS18B20 temperature sensor.

This is a 1-Wire sensor, so it depends on the [onewire_bus](https://components.espressif.com/components/espressif/onewire_bus) component.

### CODE REVIEW

**Comment by suda-morris** [comment]
Thanks @tore-espressif Now the CI passed.

@espzav could you please help review as well?

File: components/ds18b20/src/ds18b20.c [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    // send command: DS18B20_CMD_WRITE_SCRATCHPAD
+    uint8_t tx_buffer[10] = {0};
+    tx_buffer[0] = ONEWIRE_CMD_MATCH_ROM;
+    memcpy(&tx_buffer[1], &ds18b20->addr, sizeof(ds18b20->addr));
+    tx_buffer[9] = DS18B20_CMD_WRITE_SCRATCHPAD;
+    ESP_RETURN_ON_ERROR(onewire_bus_write_bytes(ds18b20->bus, tx_buffer, 10), TAG, "send DS18B20_CMD_WRITE_SCRATCHPAD failed");
+
+    // write new resolution to scratchpad
+    tx_buffer[0] = ds18b20->th_user1;
+    tx_buffer[1] = ds18b20->tl_user2;
```
Comment (espzav) [inline components/ds18b20/src/ds18b20.c:86]: I don't know the protocol for it, but the variables `ds18b20->th_user1`, `ds18b20->th_user2` was never changed, is it right?

**Review by tore-espressif — APPROVED** [review]
I could not check on HW, but the code LGTM

File: components/ds18b20/include/ds18b20_types.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+/**
+ * @brief DS18B20 supported resolutions
+ */
+typedef enum {
+    DS18B20_RESOLUTION_9B,  /*!<  9bit, needs ~93.75ms convert time */
+    DS18B20_RESOLUTION_10B, /*!< 10bit, needs ~187.5ms convert time */
+    DS18B20_RESOLUTION_11B, /*!< 11bit, needs ~375ms convert time */
+    DS18B20_RESOLUTION_12B, /*!< 12bit, needs ~750ms convert time */
+} ds18b20_resolution_t;
```
[the remark this answers is not in the document]
Comment (suda-morris) [reply components/ds18b20/include/ds18b20_types.h:20]: No special reason I'm sorry. Personally I like to put general type defines in a single file and put function declaration, configuration structures in another file.

File: components/ds18b20/src/ds18b20.c (hunk 1 above)
Comment (suda-morris) [reply components/ds18b20/src/ds18b20.c:86]: yes, this is part of the **not implemented yet** features (the reason why this component is just 0.1.0).

**Comment by suda-morris** [comment]
@tore-espressif @espzav thanks for your review and useful comments! I've addressed them all. Please help merge if it looks okay now. This component will be used by one of esp-idf's example.

**Review by espzav — APPROVED** [review]
Thank you for all changes! LGTM

**Pull request merged**